### PR TITLE
misc: add direnv and a nix devshell

### PR DESCRIPTION
**Pull request by jrobsonchase**
Add a .envrc and flake.nix to provision development tooling via nix and
direnv.

### CODE REVIEW

**Comment by jrobsonchase** [comment]
Current dependencies on/for this PR:
* main
  * **PR #227** <a href="https://app.graphite.dev/github/pr/ngrok/kubernetes-ingress-controller/227" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/ngrok/kubernetes-ingress-controller/227?utm_source=stack-comment).

File: .envrc [hunk 1]
```diff
@@ -0,0 +1,6 @@
+if ! has nix_direnv_version || ! nix_direnv_version 2.2.0; then
+  source_url "https://raw.githubusercontent.com/nix-community/nix-direnv/2.2.0/direnvrc" "sha256-5EwyKnkJNQeXrRkYbwwRBcXbibosCJqyIUuz9Xq+LRc="
+fi
+GOPATH=`pwd`/.direnv/go
+PATH=${GOPATH}/bin:${PATH}
```
Comment (nikolay-ngrok) [inline .envrc:5]: wouldn't these get added by nix/flake?

File: .envrc [hunk 2]
```diff
@@ -0,0 +1,6 @@
+if ! has nix_direnv_version || ! nix_direnv_version 2.2.0; then
+  source_url "https://raw.githubusercontent.com/nix-community/nix-direnv/2.2.0/direnvrc" "sha256-5EwyKnkJNQeXrRkYbwwRBcXbibosCJqyIUuz9Xq+LRc="
```
Comment (nikolay-ngrok) [inline .envrc:2]: are we pinning `direnv`'s version here?

Comment (jrobsonchase) [reply .envrc:2]: I don't actually remember where this came from :sweat_smile: 

I came back to the repo with a half-created direnv/flake env and went with it lol. I'll get rid of this bit.

File: .envrc (hunk 1 above)
Comment (jrobsonchase) [reply .envrc:5]: No, the flake can't refer to the original source location - only nix store locations after the flake is copied there.

Comment (nikolay-ngrok) [reply .envrc:2]: oh this is so we can completely isolate go workings to the repository?

**Review by alex-bezek — APPROVED** [review]
We don't have a ton of docs yet, but nix might be worth adding a note about to the development guide since its not well known https://github.com/ngrok/kubernetes-ingress-controller/tree/main/docs/developer-guide at some point

File: .envrc (hunk 1 above)
Comment (jrobsonchase) [reply .envrc:2]: Yeah, mainly for things like automatically `go install`'d tools via the IDE

**Comment by bobzilladev** [comment]
Looks like `helm` is an audio synthesizer, we probably want `kubernetes-helm`?
https://github.com/NixOS/nix/issues/3283
```
error: Package ‘helm-0.9.0’ in /nix/store/xl3m1kfiqmmdrd50imsw6h3g6s8nxrsj-source/pkgs/applications/audio/helm/default.nix:52 is not available on the requested hostPlatform:
         hostPlatform.config = "aarch64-apple-darwin"
```

**Pull request merged**